### ISSUE TITLE
nn.Linear / F.linear doesn't check for input shapes and instead throws CUDA OOM

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#67839]
The shapes are clearly incompatible, but instead of shape error it tries to allocate something big (torch 1.9.1)

```python
import torch

input = torch.rand(2048, 256, 7, 7).cuda()
weight = torch.rand(4096, 9216).cuda()
bias = torch.rand(4096).cuda()

torch.nn.functional.linear(input, weight, bias)

#     return torch._C._nn.linear(input, weight, bias)
# RuntimeError: CUDA out of memory. Tried to allocate 56.00 GiB (GPU 0; 10.76 GiB total capacity; 242.02 MiB already allocated; # 9.51 GiB free; 244.00 MiB reserved in total by PyTorch)
```

**Comment by gchanan** [comment]
I believe this was fixed on 1.10 -- do you mind checking?

**Comment by vadimkantorov** [comment]
Can't check right now :( working towards CVPR, not the time to change versions :(

**Comment by ngimel** [comment]
Closing as it's fixed in 1.10 and on master, nothing actionable for us to do.
